apollo-router/src/plugins/rhai.rs

```diff
@@ -1500,6 +1500,14 @@ impl Rhai {
                 TraceId::maybe_new().ok_or_else(|| "trace unavailable".into())
             })
             .register_fn("to_string", |id: &mut TraceId| -> String { id.to_string() })
+            // Register basic "crypto" functions
+            // Base64 decode/encode
+            .register_fn("base64decode", |input: &mut ImmutableString| -> Result<String, Box<EvalAltResult>> {
```

Comment:
Do you think we should provide the same name than in JavaScript ? Like `btoa` and `atob` ? We could have both but I'm just suggesting this to make life easier for our users.

Reply:
I don't like the name either. I picked "urlencode()/urldecode()" for the url coding stuff and so I was trying to remain consistent with that, but I regret that naming choice now. I'd probably prefer:
 - b64_encode/b64_decode
 - base64_encode/base64_decode
I don't love the btoa/atob suggestion, but I'm prepared to be convinced.

Reply:
My preference would be: anything but `btoa`/`atob`
:sweat_smile:

Reply:
There is [namespace support. ](https://rhai.rs/book/language/fn-namespaces.html?highlight=namespace#module-namespaces). We could also do it by having a new type (base64) with encode and decode.

Probably needs some more discussion/design.

Reply:
true, happy to go with that in the meantime then!

Reply:
Proposal:

New module (namespace) "base64" with two functions, encode and decode. When/If we get around to having a crypto module, we can make the crypto module import base64, or users could still import base64 module directly.

Reply:
I don't think base-64 belongs in "crypto"

+1  for a `base64` module with `encode` and `decode` functions

Reply:
I didn't say I loved the `btoa` and `atob` version :p I just wanted to provide a more "javascript" aware naming :p . My assumption about rhai is that it will be easily compared to javascript (about the syntax). That's why I suggested to have both versions.

Reply:
I'm going to mark this resolved with the addition of the `base64` module in ed3691b1b64043ab5aeed4bf666b3508b9ee23cb .

If you disagree, resume this conversation.